person.rb

```diff
@@ -1,17 +1,24 @@
-require './nameable_class'
+require './nameable'
 
 # This is the main class of the application
 class Person < Nameable
-  attr_accessor :name, :age, :parent_permission
+  attr_reader :id, :parent_permission, :rentals
+  attr_accessor :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
 
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```

Comment:
- [ ] Here also use this :
```ruby
def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

Reply:
thank you so much for adding an example, it was constructive.